gateway/proxy.go

```diff
@@ -0,0 +1,38 @@
+package gateway
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+)
+
+type Proxy struct {
+	targetURL    *url.URL
+	reverseProxy *httputil.ReverseProxy
+}
+
+func NewProxy(targetURL string) (*Proxy, error) {
+	url, err := url.Parse(targetURL)
+	if err != nil {
+		return nil, err
+	}
+
+	reverseProxy := httputil.NewSingleHostReverseProxy(url)
+	originalDirector := reverseProxy.Director
+	reverseProxy.Director = customDirector(url, originalDirector)
+
+	return &Proxy{
+		targetURL:    url,
+		reverseProxy: reverseProxy,
+	}, nil
+}
+
+func customDirector(targetURL *url.URL, originalDirector func(*http.Request)) func(*http.Request) {
+	return func(r *http.Request) {
+		originalDirector(r)
+	}
```

Comment:
I wrote this customDirector thinking that we would need it in the future

Reply:
I think is important you put that comment in the code, it will help your future "you" understand why you did it 😄